gtsam/inference/Factor.h

```diff
@@ -25,6 +25,7 @@
 #include <boost/serialization/nvp.hpp>
 #endif
 #include <memory>
+#include <algorithm>
```

Comment:
Do we need this?

Reply:
Yeah to fix latest clang 15 on Linux, I tested on my arch. (we are using `std::find`)